### ISSUE TITLE
Design Mockup

### ISSUE THREAD

**Issue by jonycoo** [hospitalityclub/hc-app#1]
I have done some work to show you the first sketch of the Design, if you have have any idea how to improve it please write an answer.

<h3>Main</h3>

![main symbol design](https://user-images.githubusercontent.com/16562549/52995752-d7c0aa00-341b-11e9-9ad7-6bdcb477ed4f.png)

<h3>Login</h3>

![login](https://user-images.githubusercontent.com/16562549/52995587-5b2dcb80-341b-11e9-99a4-1034a83d52be.png)

<h3>Profile Set</h3>

![profile set tab design](https://user-images.githubusercontent.com/16562549/52995815-0474c180-341c-11e9-8531-c603f7de70d5.png)

<h3>Side Bar</h3>

![sidebar](https://user-images.githubusercontent.com/16562549/52995951-5fa6b400-341c-11e9-8e40-315264e73201.png)

**Comment by infohata** [comment]
This is very good guys, love it =) this is how UI and UX work should start.

My input concerns modifications to the data model and things needing an overhaul.

First and foremost - strings will come from the backend. They are stored in `_("")` expressions in `hc4/hccore/models.py` file. All strings will be translatable through gettext.

The icon grid looks fine, but I think we should implement it more or less into a dashboard.
If a user is not logged in, in the app user should be able to completely browse the website's informational/guideline resources, and of course login/logout. These resources should also be available once logged in, may be accessible through the burger menu.

Sign up form - I agree to separate it to pages, and there is a good reason to do so.
Contact Methods - these will be dynamic and while we can template some, most of the user will be able to add/remove. Also, each contact method must have 2 checkboxes - one is "preferred", another is "share by default". @vkuehne maybe i termed it wrong, but share by default would function that the contact detail is getting shared once accepted hosting request or when a person is traveling and applying to be a guest.

Addresses are now also dynamic. While there will always be the primary address, you can have multiple, and switch between them (selecting which is primary at the time of relevance).

Also some fields will be translatable, for example - street address could be provided in native and Egnlish, and any other relevant language. This would especially be relevant for localities with non-latin alphabet of the native language. Also please consider absolute multilingualism in your app UI.

So far my feedback, once again, amazing work!!

**Comment by vkuehne** [comment]
Yeah, I was also veeeery happy when @jonycoo sent me these mockups a few days ago - I'll use them soon in another post to see if we can create a bit more buzz around this project and find more (app) developers.

I only have a couple of minor fixes at the moment: Hospitality Club please always separated, so space in between the words. And Google+ is dying afaik, so probably shouldnt have it on the mockups either.

@infohata: contact methods - I think so far our strategy of hiding the required info if wanted (name, address, email automatically hidden), and then letting members add whatever else they want in their profile worked quite well, no need to overcomplicate it. This will be a big difference to CS where they try to force members to use the site, and ban events just for adding a whatsapp number :D. We probably will have a few more new datasets (mobile phone, FB login) that we'll also encourage people to enter (for verification etc), and those should also freely hideable/unhideable.

Your functionality of automatically sending info to future guests or hosts is great, but should be built into the hosting request system, so more stuff like address, directions, further instructions can be included with a click. We do the same already with full name+passport number with messages/hosting requests.

**Comment by infohata** [comment]
It is still better to have one abstract implementation and manage it politically, than to have extra 20 fields in the database and maintain them for APIs, UI and so on. I've implemented it already in the data model in my last commit [here](https://github.com/hospitalityclub/hc-django/commit/87aeaa97958898924f5e6d5e1370951eba243d19#diff-85a3c87b58b14cd30336a1cb42354b2fR239) - have a look

**Comment by infohata** [comment]
so you will be able to add as many phone numbers, emails, skypes and telegram handles or facebook links as you want.. 

authorizing account with facebook can be done through oauth, but i'd not focus on it now, even if it is not too technically complicated, it involves opening and maintaining facebook app, which I recommend to do correctly from day 1. I think Tilman could do that effectively.

**Comment by vkuehne** [comment]
Ok, I trust you about the contact fields, just one thing - wouldnt we be able to add direct contact options automatically, depending on what contact option the user selects (so direct click on phone number opens whatsapp etc.). Anyway, not so important right now.

FB login: that's actually something I'd really love from the get-go. This way members could login without remembering their old pw (and for some of them, it will have been years and years since they did), and we could also do fun things like have them connect to their FB friends on HC. And of course it would already be a very good additional Verification step, due to most people not using FB with fake names etc. anymore.

Similar with Instagram, btw. While I f*****ing hate despise loathe this app, everyone is addicted to it. And if we allow people to show off their last pics in the profile like Tinder and other dating apps do, it would be a quick way to make profiles more interesting and visual right away (before we start building our own photo solutions...CS has that, although it always feels a bit clumsy).

So different contact fields might have different usages/ways they get integrated into the site. Maybe Telegram has some chat-box where you can chat with the member right away, dunno...but I'm sure there will be lots of cool things to do with the contact ways members chose to put into their profiles.

P.S.: I'll ask Tilman to check here.

**Comment by tilmanba** [comment]
Uff, I had to create an account on this thing called Github just to ask a question: What does "authorizing account with facebook" mean???

**Comment by vd-l** [comment]
@Tilman, Aha, that's very funny! :-)

I've used Facebook Login (OAuth) and related features in the past in
Perl, without an API, so I guess I would be able to implement it in
Python where an SDK actually exists. I've never developped Android
apps but I should be able to help there as well. I can't help with iOS
though.

If you'll want me to help, let me know when and, if wanted, provide
specifications. Or don't give me specification if you prefer, I'm
familiar enough with the HC database that I'll be able to figure
something out on my own.

@Veit, Do note that using Facebook Login will only be straightforward
if people used the same email address for their HC and FB accounts. It
may be also useful to think of a form where people could fill their
names, address, etc. if they don't remember either their HC username
or the email address they used—FB Login could help them fill this form
more quickly. And if they can't access this email address anymore and
don't remember their password, well, you'll have to figure whether you
want to compromise security by allowing them to recover their account
with some sort of challenge, or if you prefer to compromise security
by having them create a new account. :-D

VDL

**Comment by vkuehne** [comment]
@vd-l : Dude, who the f**k are you? You get exactly what we're doing here :) I'll reply on the issues at hand when I'm ....well, sober :)

**Comment by vd-l** [comment]
Oh, aha, sorry! I've configured GitHub. It should include my name now. :-)

Valmi

**Comment by infohata** [comment]
Hello guys, let's address things all at the time to avoid redundant effort.

@tilmanba "authorizing account with facebook" means basically "login with facebook"
@vd-l - new data model is being developed and "initial migration" is ready to be produced, and the tool to import old data is in cooking as I am writing. Please check [this branch](https://github.com/hospitalityclub/hc-django/blob/hc-3-core-app-data-model/hc4/hccore/models.py) of hc-django repo. And yes, even Django has OAuth module ready to plug in. As I've said, it's more of a political job than technical to implement it, and first, we need both the new site and the facebook app ready set up and running.

**Comment by infohata** [comment]
Concerning recovery, I have some wild ideas, please formally report the challenge in a new issue in the hc-django repo.

**Comment by vkuehne** [comment]
First of all: welcome back, Valmi!! Awesome to get your input here - looooove it :) If you want to join our Telegram developer chat, just let one of us know to add you there!

I just opened a new issue for the FB login app and will try to copy+paste everything relevant over there: https://github.com/hospitalityclub/hc-django/issues/8

**Comment by vkuehne** [comment]
also, as requested by @infohata , here's the recovery ticket, already put your thoughts in there: https://github.com/hospitalityclub/hc-django/issues/9

**Comment by jonycoo** [comment]
Hey, a short update:
@vkuehne proposed a chance of the Main View to a Dashboard like in the Couchsurving App.
Here is how that could look like.

![main dashboard](https://user-images.githubusercontent.com/16562549/53243172-5aa96500-36a7-11e9-889f-a33c28f92981.png)

**Comment by jonycoo** [comment]
what do you think? Stick with the Symbol Main or use a Dashboard and a Tapped Design?

**Comment by jonycoo** [comment]
Will delete this comment

**Comment by vkuehne** [comment]
Uh, I hope I didn't say "Dashboard like in CS" - actually don't like that word at all, and not a big fan of that app either.

I think with the app main screen we want to achieve 2 things: 
- quick access to the most important HC features (profile, map, city page, messages,...)
- activity/news/content - since the community will not be superactive at the very beginning, we need to find interesting/changing stuff to display there so people use the app and don't give up after one try. Of course any received messages, comments, profile visitors, prompts to complete profile, later also guests coming to an area, groups a member joined with their last posts, activity of friends. At the beginning maybe also more general stuff like "last comments posted on HC", or other crazy or fun stuff we come up with :)

How we best achieve this - no idea, trust you to find the right mix design-wise. And we'll use trial and error to further improve the app of course.

**Comment by jonycoo** [comment]
Added a Mail Design uploaded most mock ups as png including discarded once.
Rearranged main view to fulfill requirements issued in previous https://github.com/hospitalityclub/hc-app/issues/1#issuecomment-466620312
 commit: https://github.com/hospitalityclub/hc-app/commit/db0274594e333284fe239371ec8607c97e6c5de1
